docs/source/en/conceptual/reproducibility.mdx

```diff
@@ -0,0 +1,159 @@
+
+```python
+import torch
+from diffusers import DDIMPipeline
+import numpy as np
+
+model_id = "google/ddpm-cifar10-32"
+
+# load model and scheduler
+ddim = DDIMPipeline.from_pretrained(model_id)
+
+# create a generator for reproducibility
+generator = torch.Generator(device="cpu").manual_seed(0)
+
+# run pipeline for just two steps and return numpy tensor
+image = ddim(num_inference_steps=2, output_type="np", generator=generator).images
+print(np.abs(image).sum())
+```
+
+Running the above always prints a value of 1491.1711 - also upon running it again because we 
+define the generator object to be passed to all random functions of the pipeline.
+
+If you run this code snippet on your specific hardware and version, you should get a very similar, if not the same result.
+
+<Tip>
+
+It might be a bit unintuitive at first to pass "generator" objects to the pipelines instead of 
+just integer values representing the seed, but this is the recommended design when dealing with 
+probabilistic models in PyTorch as generators are *random states* that are advanced and can thus be 
+passed to multiple pipelines in a sequence.
+
+</Tip>
+
+Great! Now, we know how to write reproducible pipelines, but it does get a bit tricker since the above example only runs on CPU. How do we also achieve reproducibility on GPU? 
+In short, one should not expect to get full reproducibility across hardwares when running pipelines on GPU 
+as matrix multiplications are less deterministic on GPU than on CPU and diffusion pipelines tend to require
+a lot of matrix multiplications. Let's see what we can do though to keep the randomness within limits across 
+different GPU hardware.
+
+To achieve maximum speed performance, it is recommended to create the generator directly on GPU when running 
+the pipeline on GPU:
+
+```python
+import torch
+from diffusers import DDIMPipeline
+import numpy as np
+
+model_id = "google/ddpm-cifar10-32"
+
+# load model and scheduler
+ddim = DDIMPipeline.from_pretrained(model_id)
+ddim.to("cuda")
+
+# create a generator for reproducibility
+generator = torch.Generator(device="cuda").manual_seed(0)
+
+# run pipeline for just two steps and return numpy tensor
+image = ddim(num_inference_steps=2, output_type="np", generator=generator).images
+print(np.abs(image).sum())
+```
+
+Running the above now prints a value of 1389.8634 - even though we're using the exact same seed! 
+This is unfortunate as it means we cannot really reproduce results we achieved on GPU, also on CPU.
+Nevertheless, it should be expected since the GPU uses a different random number generator than the CPU.
+
+To circumvent this problem, we created a [~`randn_tensor`] function which can create random noise 
+on CPU and then move the tensor to GPU if necessary. The function is used every where inside the pipelines allowing the user to **always** pass a CPU generator even if the pipeline is run on GPU:
```

Comment:
```suggestion
Running the above now prints a value of 1389.8634 - even though we're using the exact same seed! 
This is unfortunate as it means we cannot reproduce the results we achieved on GPU, also on CPU.
Nevertheless, it should be expected since the GPU uses a different random number generator than the CPU.

To circumvent this problem, we created a [~`randn_tensor`] function, which can create random noise 
on the CPU and then move the tensor to GPU if necessary. The function is used everywhere inside the pipelines allowing the user to **always** pass a CPU generator even if the pipeline is run on GPU:
```